### ISSUE TITLE
Gap is heard on gapless album on Chromecast player

### ISSUE THREAD

**Issue by madbrain76** [music-assistant/hass-music-assistant#2901]
### What version of Music Assistant has the issue?

2.3.0b19

### What version of the Home Assistant Integration have you got installed?

None

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

There is a gap between tracks playing a gapless album on a Chromecast audio . This is not related to track collapsing

### How to reproduce

1. Add the Qobuz music provider
2. Load the album "Handel, G.F.: Keyboard suites Nos 1, 3, 4, 5 and 7 / Chaconne, Hwv 435" by Byron Schenkman
3. Select a Chromecast player with "queue flow" on . Volume normalization is also off
4. Play the album from the start
5. A long gap is heard between the end of track 1 and the start of track 2 . I would say over 1 second

Edit: I recorded the output of my CCA to a WAV file. The gap is indeed just over 1 second.

![image](https://github.com/user-attachments/assets/44f458a0-7004-4e74-9563-0d750fc3f3df)

I have saved the WAV file, in case it is needed.

### Music Providers

Qobuz

### Player Providers

Chromecast

### Full log output

[Uploading ma_gap.log…]()

### What version of Home Assistant Core are your running

2024.9.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Windows

**Comment by madbrain76** [comment]
I tried on an Airplay player (WiiM Pro Plus) and did not hear the gap.

**Comment by OzGav** [comment]
Can you just try again with beta 21 as I just tried some tracks with flow mode on and didn't notice a 1 second gap?

**Comment by madbrain76** [comment]
@OzGav , it still happens on beta 21 with Qobuz and the album I mentioned.

https://play.qobuz.com/album/0044747243627

**Comment by OzGav** [comment]
I don't have Qobuz to test unfortunately. What if you play other albums, do you notice gaps? I notice a slight pause with Chromecast if the queue flow mode is off but nothing when it is on? Do you have another Chromecast device as I am testing with a Google Home?

**Comment by madbrain76** [comment]
@OzGav,

Not sure if you have any streaming service, but if so, the album is likely on it. It is usually called "Harpsichord variations" . That is what it says on my CD cover.

https://www.amazon.com/Harpsichord-Variations-DAVID-DIAMOND/dp/B00004T00I/ref=sr_1_1?dib=eyJ2IjoiMSJ9.UwYau7ePNog2PCJQr984BMJoF-A_IXjaKsACgnGmTGLSA87vsEkPW1YKxgMHREwv3nID8dvAC2WUCIalZYs8EhqamiuSubnTtbeL1PYvFO7vdWlYHM8X0YUxaUy1e-Ffh8KRy9Zn_PbeI1ZcYFGmXAKYNMMLFkMt_JnqZxf27uIgz7kt45dPbcYh69MHICgoELlJydO3GJ4hStJEYNvig8wkmlped2Rt_dK-LZh9YWg.jLl3YZ4s0DoNr5rNILrBgTkJ72nt9aj6G2RI_MzAOd0&dib_tag=se&keywords=byron+schenkman+handel&qid=1726108131&sr=8-1

I do have access to another Chromecast. I checked with a WiiM Pro Plus built-in CCA ... And experienced no gap. Weird. That was just the one time. I'll let the album play through. It's gapless throughout, for nearly all tracks. It's on track 3 now, and no gap.

According to the Google Home app, the Chromecast audio in my office which is experiencing the issue is on Cast firmware 1.56.281627 . According to https://support.google.com/chromecast/answer/7124014?hl=en#zippy=%2Ccurrent-preview-program-firmware-version , 1.56.281627 is the latest version for the Chromecast audio.

The WiiM Pro Plus not experiencing the issue is on cast firmware 1.56.310669 .

**Comment by madbrain76** [comment]
Just checked, and it's also a problem on a Chromecast ultra .

**Comment by madbrain76** [comment]
No issue on Chromecast with Google TV (4K version).

**Comment by madbrain76** [comment]
I installed the "AirCast" HAOS add-on, which exposes an AirPlay device for each Chromecast device. I was then able to play gaplessly on my CCA using the Airplay provider in MA.

**Comment by marcelveldt** [comment]
Chromecast is NOT gapless - enable flow mode if you want it to be gapless

**Comment by madbrain76** [comment]
Marcel,

I see indeed that the Office CCA does not have queue flow mode turned on. This doesn't make any sense to me. When I first installed the beta, I turned queue flow mode on every single CC device . I don't know what happened since.
I will check my backups to see when this might have been changed.

**Comment by madbrain76** [comment]
I checked my backups . I went from beta 8 to 15, 16 and 17 . The queue flow is disabled on the Office CCA starting at the backup made with b17 . I have no idea how it got disabled. I certainly did not do it intentionally.

For the Chrromecast ultra device, I had forgotten to enable it from the beginning.
That device is normally powered off by smartplug, and it's not possible to configure it while it's off, which probably explains why I forgot.
